suite/auto-sync/Update-Arch.sh

```diff
@@ -111,62 +112,71 @@ tblgen="$2"
-"$tblgen" --printerLang=CCS --gen-register-info -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td" > $llvm_c_inc_dir"/"$arch"GenRegisterInfo.inc"
+"$tblgen" --printerLang=CCS --gen-register-info -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td" > $llvm_c_inc_dir"/""$arch""GenRegisterInfo.inc"
 verify_status $? "[X] Failed to generate RegisterInfo tables for capstone"
 
 echo "[*] Generate InstrInfo tables..."
-"$tblgen" --printerLang=CCS --gen-instr-info -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td" > $llvm_c_inc_dir"/"$arch"GenInstrInfo.inc"
+"$tblgen" --printerLang=CCS --gen-instr-info -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td" > $llvm_c_inc_dir"/""$arch""GenInstrInfo.inc"
 verify_status $? "[X] Failed to generate InstrInfo tables for capstone"
 
 echo "[*] Generate SubtargetInfo tables..."
-"$tblgen" --printerLang=CCS --gen-subtarget -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td" > $llvm_c_inc_dir"/"$arch"GenSubtargetInfo.inc"
+"$tblgen" --printerLang=CCS --gen-subtarget -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td" > $llvm_c_inc_dir"/""$arch""GenSubtargetInfo.inc"
 verify_status $? "[X] Failed to generate SubtargetInfo tables for capstone"
 
+cd "$llvm_c_inc_dir" || print_and_pause "[X] Could not enter $llvm_c_inc_dir directory"
 echo "[*] Generate Mapping tables..."
 "$tblgen" --printerLang=CCS --gen-asm-matcher -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td"
 verify_status $? "[X] Failed to generate Mapping tables for capstone"
+mv $files_to_move ..
+cd ..
 
 has_sys_reg_tables="ARM, AArch64"
 if echo "$has_sys_reg_tables" | grep -q -w "$arch"; then
+  cd "$llvm_c_inc_dir" || print_and_pause "[X] Could not enter $llvm_c_inc_dir directory"
   echo "[*] Generate System Register tables..."
   "$tblgen" --printerLang=CCS --gen-searchable-tables -I "$path_to_llvm/include/" -I "$path_to_llvm/lib/Target/$llvm_target_dir/" "$path_to_llvm/lib/Target/$llvm_target_dir/$arch.td"
   verify_status $? "[X] Failed to generate System Register tables for capstone"
   sed -i "s/##ARCH##/$arch/g" __ARCH__GenCSSystemRegisterEnum.inc
   sed -i "s/##ARCH##/$arch/g" __ARCH__GenSystemRegister.inc
-  cp __ARCH__GenCSSystemRegisterEnum.inc $arch"GenCSSystemRegisterEnum.inc"
-  cp __ARCH__GenSystemRegister.inc $arch"GenSystemRegister.inc"
+  mv __ARCH__GenCSSystemRegisterEnum.inc "../$arch""GenCSSystemRegisterEnum.inc"
+  mv __ARCH__GenSystemRegister.inc "$arch""GenSystemRegister.inc"
```

Comment:
This whole `__ARCH__` and `##ARCH##` replacement is outdated (it was a horrible idea to begin with). The `AArch64` tblgen version now generates the files with proper arch names. You'll see when you rebase.